app/src/main/java/com/firebase/uidemo/database/ChatActivity.java

```diff
@@ -131,26 +136,50 @@ public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
     }
 
     private void attachRecyclerViewAdapter() {
-        Query lastFifty = mChatRef.limitToLast(50);
-        mAdapter = new FirebaseRecyclerAdapter<Chat, ChatHolder>(
-                Chat.class, R.layout.message, ChatHolder.class, lastFifty) {
+        mAdapter = new FirebaseIndexRecyclerAdapter<Chat, ChatHolder>(
+                Chat.class,
+                R.layout.message,
+                ChatHolder.class,
+                mChatIndicesRef.limitToLast(50),
+                mChatRef) {
             @Override
-            public void populateViewHolder(ChatHolder holder, Chat chat, int position) {
-                holder.setName(chat.getName());
-                holder.setText(chat.getMessage());
-
-                FirebaseUser currentUser = mAuth.getCurrentUser();
-                if (currentUser != null && chat.getUid().equals(currentUser.getUid())) {
-                    holder.setIsSender(true);
-                } else {
-                    holder.setIsSender(false);
-                }
+            public void populateViewHolder(final ChatHolder holder, Chat chat, int position) {
+                holder.bind(chat);
+
+                holder.itemView.setOnCreateContextMenuListener(new View.OnCreateContextMenuListener() {
+                    @Override
+                    public void onCreateContextMenu(ContextMenu menu,
+                                                    View v,
+                                                    ContextMenu.ContextMenuInfo menuInfo) {
+                        menu.add("Delete")
+                                .setOnMenuItemClickListener(new MenuItem.OnMenuItemClickListener() {
```

Comment:
I like the idea but the ratio of "lines of code" to "how much it teaches you about FirebaseUI" may be too low here.  If you don't have a strong preference I'd say remove it to keep `ChatActivity` as short as possible.